### PR TITLE
doc(quick-start) : improvements

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- Improve the steps to create a new project
- add section next steps to make clear that demo project will be checked by following the tutorial. 
- add links to golang docs for new users which are not skilled in Go. 

**Motivation**
Closes: #1031 and #1033

### CODE REVIEW

**Comment by camilamacedo86** [comment]
HI @erictune, 

I think this PR address the missing points for we close the issues. WDYT? Has anything else that you would like to suggest here for we close the issues https://github.com/kubernetes-sigs/kubebuilder/issues/1031, https://github.com/kubernetes-sigs/kubebuilder/issues/1034, https://github.com/kubernetes-sigs/kubebuilder/issues/1033.

File: docs/book/src/quick-start.md [hunk 1]
```diff
@@ -41,17 +41,20 @@ Also, you can install a master snapshot from `https://go.kubebuilder.io/dl/lates
...
 ```
 
 <aside class="note">
 <h1>Not in $GOPATH</h1>
 
-If you're not in `GOPATH`, you'll need to run `go mod init <modulename>`
-in order to tell kubebuilder and Go the base import path of your module.  
+If you're not in `GOPATH`, you'll need to run `go mod init <modulename>` in order to tell kubebuilder and Go the base import path of your module. 
+
+For a further understand over `GOPATH` see [The GOPATH environment variable][GOPATH-golang-docs] in the [How to Write Go Code][how-to-write-go-code-golang-docs] golang page doc.
```
Comment (erictune) [inline docs/book/src/quick-start.md]: replace "further understand over " with "further understanding of"

File: docs/book/src/quick-start.md [hunk 2]
```diff
@@ -193,6 +196,11 @@ privileges or be logged in as admin. See [Prerequisites for using Kubernetes RBA
 
 </aside> 
 
-Now, follow up the tutorial for you understanding better how it works and check an full example project working. 
+## Next Step 
+
+Now, follow up the CronJob tutorial[cronjob-tutorial] for you understanding better how it works and check a demo example project working on.
```
Comment (erictune) [inline docs/book/src/quick-start.md]: "for you understanding better" ➡️ "to better understand"
"check" ➡️ "try".
delete "working on".

**Comment by erictune** [comment]
IMO this resolves #1031 and #1033 

Will follow up separately in #1034.

File: docs/book/src/quick-start.md (hunk 1 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Done.

File: docs/book/src/quick-start.md (hunk 2 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Done

**Pull request merged**